core/src/elo-connector/elo-connector.service.ts

```diff
@@ -0,0 +1,291 @@
+     * all subsequent replays where those player's played need to be recalculated.
+     * @param seriesId The series to mark full_ncp=true/false.
+     * @param isNcp Whether the given replayId should be marked NCP or un-NCP.
+     * @param progressMessage A message to update with the progress of the NCP processing.
+     * @param seriesType Whether the series is a fixture or scrim series, used for input validation and for difference in actions between fixtures/scrims.
+     * @param winningTeam The team that should win the NCP. Required if seriesType=Fixture and isNcp=true.
+     * @param numReplays The number of replays that should be in the series. Optional. Used to add dummy replays in place of replays that weren't submitted for some reason.
+     * @returns A string containing a summary of the actions that took place when the processing has completed.
+     */
+    async markSeriesNcp(seriesId: number, isNcp: boolean, seriesType: SeriesType, winningTeamInput?: Team, numReplays?: number): Promise<string> {
+        const r = Math.floor(Math.random() * 10000);
+        this.logger.verbose(`(${r}) begin markSeriesNcp: seriesId=${seriesId}, isNcp=${isNcp}, winningTeam=${winningTeamInput}`);
+
+        // Find the winning team and it's franchise profile, since that's where
+        // team names are in Sprocket.
+        let winningTeam = await this.teamRepository.findOne(winningTeamInput?.id, {where: {id: winningTeamInput?.id}, relations: ["franchise", "franchise.profile"] });
+
+        // const series = await this.ss.getSeriesById(seriesId, ["fixture.home",
+        // "fixture.away"]);
+        const series = await this.matchRepository.findOneOrFail(seriesId, {where: {id: seriesId}, relations: ["matchParent", "rounds"] });
+
+        if (seriesType === SeriesType.Fixture) {
+
+            if (!series.matchParent.fixture) {
+                return `Error: series with id=\`${seriesId}\` has no associated fixture. If this is a scrim, please use \`!ncpScrim\` or \`!unNcpScrim\`. If this actually is a league play series, ping Bot Team.`;
+            }
+
+            // Winning team must be specified if NCPing replays
+            if (isNcp && !winningTeam) {
+                return "When NCPing a series associated with a fixture, you must specify a winningTeam";
+            }
+
+            // Check to make sure the winning team played in the series/fixture
+            if (winningTeam
+                && series.matchParent.fixture.homeFranchise !== winningTeam.franchise
+                && series.matchParent.fixture.awayFranchise !== winningTeam.franchise) {
+                return `The team \`${winningTeam?.franchise.profile.title}\` did not play in series with id \`${series.id}\` (${series.matchParent.fixture.awayFranchise.profile.title} v. ${series.matchParent.fixture.homeFranchise.profile.title}), and therefore cannot be marked as the winner of this NCP. Cancelling process with no action taken.`;
```

Comment:
Starting to understand the string bits, but why not throw an error?

Reply:
Same here. This was literally pulled from MLEDB, and since it wasn't core to the NCP logic, I just glossed over it and made it compile with valid data.

Reply:
Now with thrown errors.